core/trino-main/src/main/java/io/trino/operator/scalar/SequenceFunction.java

```diff
@@ -120,6 +120,30 @@ private static Block fixedWidthSequence(long start, long stop, long step, FixedW
         return blockBuilder.build();
     }
 
+    private static int getLength(long start, long stop, long step)
+    {
+        if ((step > 0 && (start >= 0 || stop <= 0))
```

Comment:
I was thinking the case would be handle by `checkMaxEntry`, another same cases is `sequence(0, Long.MIN_VALUE, -1)`. 
Those two cases are added in the test `testSequence`.

Reply:
`stop - start` should overflow before you call `checkMaxEntry`. In fact it doesn't fail, because you are not using exact arithmetics. It just silently overflows: `0L - Long.MIN_VALUE` results in `Long.MIN_VALUE`. Indeed, this result is next caught by `checkMaxEntry()`. However, we should not rely on the incorrect (overflown) subtraction result. Please change the if-condition so that it has `stop < 0` AND use exact arithmetics in `checkMaxEntry((stop - start) / step + 1L)`

Reply:
> Please change the if-condition so that it has `stop < 0` AND use exact arithmetics in `checkMaxEntry((stop - start) / step + 1L)`

If we use exact arithmetics for `(stop - start) / step + 1L` then in some cases there could raise exception `throw new ArithmeticException("integer overflow")` is it as expected?

Reply:
>1. We need to handle two edge cases

There's another: `(0, Long.MAX_VALUE, 1)`. It fails on `+ 1L`.

We don't want to fail the query in neither of these cases.
This piece:
```
if ((step > 0 && (start >= 0 || stop <= 0))
                || (step < 0 && (start <= 0 || stop >= 0))) {
            return checkMaxEntry((stop - start) / step + 1L);
        }
```
is just a shortcut, and it works for certain limited values only. So we should adjust the if-condition so that the problematic overflowing cases don't fall here. They will fall through, and will be correctly handled by the code that follows after.

>2. 
And when you've adjusted the conditions, use the exact arithmetics as a sanity check. It should never fail. If it does, it's an internal bug, so no need to wrap in TrinoException.